streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -261,6 +262,16 @@ private KafkaStreams createCountStream(final String inputTopic,
         @Override
         public void run() {
             myStream.start();
+
+            try {
+                TestUtils.waitForCondition(
+                    () -> stateListener.mapStates.containsKey(KafkaStreams.State.RUNNING),
+                    "Did not start successfully after " + TestUtils.DEFAULT_MAX_WAIT_MS + " ms"
+                );
+            } catch (final InterruptedException e) {
+                if (!stateListener.mapStates.containsKey(KafkaStreams.State.RUNNING))
+                    fail("Did not start successfully");
+            }
```

Comment:
Why not use `startKafkaStreamsAndWaitForRunningState()` here?

Reply:
+1

Reply:
The streams instance already has a registered state listener above for further testing purposes, so I have to reuse that one than calling `startKafkaStreamsAndWaitForRunningState` to register a new one.

Reply:
Ahh! https://github.com/apache/kafka/pull/7500 allows you to wait for running state while still using an existing listener. But still waiting for it to get merged.

Reply:
I left a comment on #7500, basically I think we do not need two functions in IntegrationTestUtils and StreamsTestUtils.